### DISCUSSION TITLE
Custom Ruleset download and updating in-game?

### DISCUSSION THREAD

**Discussion by Tenexxt** [ppy/osu#29649] [Q&A] [answered]
Basically my idea for this is that near the ruleset icons you would have a button that would open an menu similar to beatmap listing.
![osu!wth](https://github.com/user-attachments/assets/1dec7140-8557-4623-8651-81060996e960) 
- how the icon would somewhat look near the rest.

from the Ruleset menu you could download and update your downloaded rulesets like beatmaps, and the github page for each ruleset would be linked.
In a rulesets page, you could see its description, which would contain how the ruleset works, and also you would have a list of supported platforms.
This would allow for mobile users to download CR's without leaving the game, and would open for a possibility of more rulesets that are made/compatible with mobile/touch devices, and could open up more people to CR's - which is a very good and possibly even needed thing in my opinion. I remeber that Peppy said that something should be done with CR's so their developers don't leave, and this is a step that would make progress in that direction.

Another thing that i think this would help with, is impementing the editor for CR's.
CR's devs would have to craft the tools themselfs, but that was already obvious in my opinion. the [rush!](https://github.com/Beamographic/rush) ruleset aleady has an editor ready, and some other ones probably also have atleast a rough model of it. The thing thats stoping the custom editors is the unability to save.
Im a very rookie programmer, but i think a way to do this is that there could be a template for the CS'r to use that would work like .osz and .osu, then their developers could adjust it for their ruleset. Every CR would have a file like that, and since you would download the CR's in game, the game would already put the file where its needed and would allow to later make and save beatmaps with a editor that was crafted by devs for you to use.

Worth noting again that im not really a programmer yet, so this concept of the impementation of the editor can be impractical and imposible to execute, but even just adding the in-game download option would be a great start on popularizing CR's more.
(Adding this as a discussion, since i think this should get discussed and also i want my mistakes on this to be told to me)

**Comment by peppy** [comment, marked as answer]
There's a lot of work to be done to make this happen, but we'll eventually look at it.
